### PR TITLE
Gateio: Add unified account API endpoint support

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Adds unified account fetching. This is just a mapping implementation, I don't have unified account enabled so I am not going to attempt to add this to the wrapper just yet, until I get an account that supports it. But it at least gives people the option to use it. 

Can close if testing requirements need to be done this is just cherry picked from another branch. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [🙅] go test ./... -race 
- [🙅] golangci-lint run
- [🙅] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/gateio/gateio_types.go [hunk 1]
```diff
@@ -2689,3 +2689,37 @@ type PriceAndAmount struct {
...
+// BalanceDetails represents a user's balance details
+type BalanceDetails struct {
+	Available      types.Number `json:"available"`
+	Freeze         types.Number `json:"freeze"`
+	Borrowed       types.Number `json:"borrowed"`
+	NegativeLiab   types.Number `json:"negative_liab"`
+	FuturesPosLiab types.Number `json:"futures_pos_liab"`
+	Equity         types.Number `json:"equity"`
+	TotalFreeze    types.Number `json:"total_freeze"`
+	TotalLiab      types.Number `json:"total_liab"`
```
Comment (gloriousCode) [inline exchanges/gateio/gateio_types.go:2702]: Liab->Liabilities for the struct

File: exchanges/gateio/gateio_types.go [hunk 2]
```diff
@@ -2689,3 +2689,37 @@ type PriceAndAmount struct {
...
+	SpotInUse      types.Number `json:"spot_in_use"`
```
Comment (gloriousCode) [inline exchanges/gateio/gateio_types.go:2703]: Missing `funding` and `funding_version`, example json doesn't have it, but you can 🚓

File: exchanges/gateio/gateio_types.go [hunk 3]
```diff
@@ -2689,3 +2689,37 @@ type PriceAndAmount struct {
...
+	TotalMaintenanceMargin     types.Number                     `json:"total_maintenance_margin"`
+	TotalInitialMarginRate     types.Number                     `json:"total_initial_margin_rate"`
+	TotalMaintenanceMarginRate types.Number                     `json:"total_maintenance_margin_rate"`
+	TotalAvailableMargin       types.Number                     `json:"total_available_margin"`
+	UnifiedAccountTotal        types.Number                     `json:"unified_account_total"`
+	UnifiedAccountTotalLiab    types.Number                     `json:"unified_account_total_liab"`
+	UnifiedAccountTotalEquity  types.Number                     `json:"unified_account_total_equity"`
+	Leverage                   types.Number                     `json:"leverage"`
+	SpotOrderLoss              types.Number                     `json:"spot_order_loss"`
+	SpotHedge                  bool                             `json:"spot_hedge"`
```
Comment (gloriousCode) [inline exchanges/gateio/gateio_types.go:2724]: Liab->Liabilities.

File: exchanges/gateio/gateio.go [hunk 4]
```diff
@@ -552,6 +552,16 @@ func (g *Gateio) GetSpotAccounts(ctx context.Context, ccy currency.Code) ([]Spot
...
 }
 
+// GetUnifiedAccount retrieves unified account.
+func (g *Gateio) GetUnifiedAccount(ctx context.Context, ccy currency.Code) (*UnifiedUserAccount, error) {
+	params := url.Values{}
+	if !ccy.IsEmpty() {
+		params.Set("currency", ccy.String())
+	}
+	var response UnifiedUserAccount
+	return &response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, spotPrivateEPL, http.MethodGet, "unified/accounts", params, nil, &response)
```
Comment (gloriousCode) [inline exchanges/gateio/gateio.go:562]: Put path with others

File: exchanges/gateio/gateio_types.go [hunk 5]
```diff
@@ -2689,3 +2689,37 @@ type PriceAndAmount struct {
...
+	NegativeLiab   types.Number `json:"negative_liab"`
+	FuturesPosLiab types.Number `json:"futures_pos_liab"`
+	Equity         types.Number `json:"equity"`
+	TotalFreeze    types.Number `json:"total_freeze"`
+	TotalLiab      types.Number `json:"total_liab"`
+	SpotInUse      types.Number `json:"spot_in_use"`
+}
+
+// UnifiedUserAccount represents a unified user account
+type UnifiedUserAccount struct {
```
Comment (gloriousCode) [inline exchanges/gateio/gateio_types.go:2707]: Missing `use_funding`, missing `refresh_time`

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Maybe by 2035 exchanges will catch up to FTX and support unified margin by default

**Review by gloriousCode — APPROVED** [review]
[ticky tACKY](https://www.youtube.com/watch?v=hNmTAxMLCZg)

**Review by thrasher- — APPROVED** [review]
All fields match docs, utACK 👍

**Pull request merged**